### PR TITLE
Add Series.is_unique property

### LINKED ISSUE #305: Series.is_unique
Although an expensive function, this should be pretty easy to add. We can do a count distinct and compare it with the total cardinality.

pandas doc: http://pandas.pydata.org/pandas-docs/stable/reference/api/pandas.Series.is_unique.html

### PR DESCRIPTION
**Pull request by HyukjinKwon**
Resolves #305

### CODE REVIEW

File: databricks/koalas/series.py [hunk 1]
```diff
@@ -348,6 +348,25 @@ def index(self):
...
+        Returns
+        -------
+        is_unique : boolean
+
+        >>> ks.Series([1, None, None]).is_unique
+        False
+        >>> ks.Series([1, 2, 3, None]).is_unique
+        True
+        """
+        sdf = self._kdf._sdf.select(self._scol)
```
Comment (rxin) [inline databricks/koalas/series.py]: the way i'd do this is to do a single pass, e.g.

select(count(col), countDistinct(col))

and get the two values back and compare them at once.

Comment (HyukjinKwon) [reply databricks/koalas/series.py]: I actually tried one pass first. The problem is countDistinct ignore null and `expr` with COUNT(DISTINCT *) ignore null by design as well.

Comment (rxin) [reply databricks/koalas/series.py]: It seems like we are fine?

```
>>> ks.DataFrame([1, 2, 3, 4, None, None]).to_spark().selectExpr("count(distinct `0`)").show()
+-----------------+
|count(DISTINCT 0)|
+-----------------+
|                4|
+-----------------+
```

This is the same as pandas:
```
>>> pd.Series([1, 2, 3, 4, None, None]).nunique()
4
```

Comment (HyukjinKwon) [reply databricks/koalas/series.py]: Yes, I will stick to the original one.

File: databricks/koalas/series.py [hunk 2]
```diff
@@ -348,6 +348,25 @@ def index(self):
             raise KeyError('Currently supported only when the Column has a single index.')
         return self._kdf.index
 
+    @property
+    def is_unique(self):
+        """
+        Return boolean if values in the object are unique
+
+        .. note:: this is an expensive operation that has to count all values in the Series.
```
Comment (rxin) [inline databricks/koalas/series.py]: i don't know if you need this note here ... of course it has to count all the values :)

File: databricks/koalas/namespace.py [hunk 3]
```diff
@@ -33,27 +33,29 @@
 from databricks.koalas.series import Series
 
 
-def from_pandas(pdf):
+def from_pandas(pobj: Union[pd.DataFrame, pd.Series]):
```
Comment (HyukjinKwon) [inline databricks/koalas/namespace.py]: Ah, I missed you submitted a PR already about this  @rxin. Sorry, let me revert

File: databricks/koalas/series.py [hunk 4]
```diff
@@ -355,6 +355,31 @@ def index(self):
...
+
+        Returns
+        -------
+        is_unique : boolean
+
+        >>> ks.Series([1, 2, 3]).is_unique
+        True
+        >>> ks.Series([1, 2, 2]).is_unique
+        False
+        >>> ks.Series([1, 2, 3, None]).is_unique
```
Comment (rxin) [inline databricks/koalas/series.py]: this should be true right?

Comment (HyukjinKwon) [reply databricks/koalas/series.py]: Actually, it gives `False` in my local:

```
>>> pd.Series([1, 2, 3, None]).is_unique
False
```

Maybe version difference. let me check

File: databricks/koalas/series.py (hunk 1 above)
Comment (HyukjinKwon) [reply databricks/koalas/series.py]: Actually, Pandas seems using:

```
>>> pd.Series([1, 2, 3, 4, None, None]).nunique(dropna=False)
5
```

**Pull request merged**